### PR TITLE
JwtWebApiSettings to support WebRoutinenBase with JWT

### PR DESCRIPTION
**Pull request by rafalgradziel**

### CODE REVIEW

File: Gandalan.IDAS.WebApi.Client/BusinessRoutinen/WebRoutinenBase.cs [hunk 1]
```diff
@@ -83,6 +91,11 @@ protected virtual void OnErrorOccured(ApiErrorArgs e)
         /// <returns>Status des Logins</returns>
         public bool Login()
         {
+            if (IsJwt)
```
Comment (gdl-reif) [inline Gandalan.IDAS.WebApi.Client/BusinessRoutinen/WebRoutinenBase.cs:94]: Validity check neccessary for expired tokens?

**Review by gdl-reif — COMMENTED** [review]
See comment

File: Gandalan.IDAS.WebApi.Client/BusinessRoutinen/WebRoutinenBase.cs (hunk 1 above)
Comment (rafalgradziel) [reply Gandalan.IDAS.WebApi.Client/BusinessRoutinen/WebRoutinenBase.cs:96]: Added JWT basic token validation (expired?) and call to `api/LoginJwt/Refresh` if `refreshToken` is available

File: Gandalan.IDAS.WebApi.Client/BusinessRoutinen/WebRoutinenBase.cs [hunk 2]
```diff
@@ -664,6 +694,123 @@ private void SetupRestRoutinen(RESTRoutinen cl)
...
+                refreshTokenGuid == Guid.Empty)
+            {
+                // JWT is expired and has no refreshToken
+                return false;
+            }
+
+            try
+            {
+                // refresh JWT using refresh token
+                //var refreshToken = refreshTokenClaim.Value;
```
Comment (Saibamen) [inline Gandalan.IDAS.WebApi.Client/BusinessRoutinen/WebRoutinenBase.cs:727]: Please delete this

File: Gandalan.IDAS.WebApi.Client/Gandalan.IDAS.WebApi.Client.csproj [hunk 3]
```diff
@@ -22,6 +22,7 @@
 	<PackageReference Include="Microsoft.CSharp" Version="4.7.0" />
     <PackageReference Include="Newtonsoft.Json" Version="13.0.1" />
     <PackageReference Include="PropertyChanged.Fody" Version="4.0.3" PrivateAssets="All" />
+    <PackageReference Include="System.IdentityModel.Tokens.Jwt" Version="6.23.1" />
```
Comment (Saibamen) [inline Gandalan.IDAS.WebApi.Client/Gandalan.IDAS.WebApi.Client.csproj:25]: Please add `System.IdentityModel.Tokens.Jwt` to Nuspec file

File: Gandalan.IDAS.WebApi.Client/BusinessRoutinen/WebRoutinenBase.cs [hunk 4]
```diff
@@ -664,6 +694,123 @@ private void SetupRestRoutinen(RESTRoutinen cl)
             }
         }
 
+        private bool CheckJwtToken()
```
Comment (Saibamen) [inline Gandalan.IDAS.WebApi.Client/BusinessRoutinen/WebRoutinenBase.cs:697]: The only difference between `CheckJwtToken` and `CheckJwtTokenAsync` is line with `var newJwt`.
Please reduce code duplication

**Review by Saibamen — CHANGES_REQUESTED** [review]
Missing changes in nuspec file

**Review by Saibamen — APPROVED** [review]

**Pull request merged**